src/humancellatlas/data/metadata/helpers/schema_validation.py

```diff
@@ -25,18 +32,28 @@ class SchemaValidator:
 
     def validate_json(self, file_json: JSON, file_name: str):
         try:
-            schema = self._download_schema(file_json['describedBy'])
+            schema = self._download_json_file(file_json['describedBy'])
         except json.decoder.JSONDecodeError as e:
             schema_url = file_json['describedBy']
             raise RequirementError('Failed to parse schema JSON',
                                    file_name, schema_url) from e
+        self.validator.evolve(schema=schema)
         try:
-            validate(file_json, schema, format_checker=FormatChecker())
+            self.validator.validate(file_json)
         except ValidationError as e:
             raise RequirementError(*e.args, file_name) from e
 
     @lru_cache(maxsize=None)
-    def _download_schema(self, schema_url: str) -> JSON:
-        response = requests.get(schema_url, allow_redirects=False)
+    def _download_json_file(self, file_url: str) -> JSON:
+        response = requests.get(file_url, allow_redirects=False)
         response.raise_for_status()
         return response.json()
+
+    def _retrieve_resource(self, resource_url: str) -> Resource:
+        file_json = self._download_json_file(resource_url)
+        return Resource.from_contents(file_json)
+
+    @cached_property
+    def validator(self) -> Draft202012Validator:
+        registry = Registry(retrieve=self._retrieve_resource)
+        return Draft202012Validator(schema={}, registry=registry, format_checker=FormatChecker())
```

Comment:
L

Please coordinate with @achave11-ucsc as he may have a PR that adds a warning suppression and FIXME for this.